middleware/realip.go

```diff
@@ -40,21 +42,28 @@ func RealIP(h http.Handler) http.Handler {
 }
 
 func realIP(r *http.Request) string {
-	var ip string
+	if tcip := r.Header.Get(trueClientIP); isValidIP(tcip) {
+		return tcip
+	}
 
-	if tcip := r.Header.Get(trueClientIP); tcip != "" {
-		ip = tcip
-	} else if xrip := r.Header.Get(xRealIP); xrip != "" {
-		ip = xrip
-	} else if xff := r.Header.Get(xForwardedFor); xff != "" {
+	if xrip := r.Header.Get(xRealIP); isValidIP(xrip) {
+		return xrip
+	}
+
+	if xff := r.Header.Get(xForwardedFor); xff != "" {
 		i := strings.Index(xff, ",")
 		if i == -1 {
 			i = len(xff)
 		}
-		ip = xff[:i]
-	}
-	if ip == "" || net.ParseIP(ip) == nil {
-		return ""
+
+		if isValidIP(xff[:i]) {
+			return xff[:i]
```

Comment:
this is slicing twice unnecessary

Reply:
```
if i := strings.Index(xff, ","); i >= 0 {
  xff = xff[:i]
}
```

Reply:
Thank you! Your contribution to refining XFF's IP address validation is very good.